web-service-consumer/1.6/modules/ROOT/pages/web-service-consumer-troubleshooting.adoc

```diff
@@ -0,0 +1,166 @@
+
+.Web Service Consumer Connector Global Element
+image::wsc-troubleshoot-3.png[Web Service Consumer global element configuration with MTOM field set to TRUE]
+
+In the XML editor, the `<wsc:config>` and `mtomEnabled` configurations look like this:
+
+[source,xml,linenums]
+----
+<wsc:config name="Web_Service_Consumer_Config" doc:name="Web Service Consumer Config">
+    <wsc:connection wsdlLocation="http://ilt.mulesoft-training.com/essentials/delta?wsdl" service="TicketServiceService" port="TicketServicePort" address="http://ilt.mulesoft-training.com/essentials/delta" mtomEnabled="true">
+        <wsc:web-service-security actor="http://schemas.xmlsoap.org/soap/actor/next" />
+    </wsc:connection>
+</wsc:config>
+----
+
+Depending on the server side multipart setting, the server may decide to send some binary data as a base64-encoded text field or as a referenced binary part in the MIME message.
+
+To troubleshoot how to access the binary data:
+
+* If the server sends the binary data as base64-encoded text field, you must access this field as part of the message. +
+Base64 decoding is not required because the Web Service Consumer performs the decoding by itself.
+* If the server sends the binary data as a referenced binary part, you must access this field as an attachment.
++
+The following example shows an image that was sent as an attachment with MTOM by a Web service server. You use DataWeave to access and transform the image into JSON. In this case, Base64 encoding is required:
++
+[source,DataWeave,linenums]
+----
+import * from dw::core::Strings
+import dw::core::Binaries
+output application/json
+ns nam http://mynamespace.com
+---
+{
+...
+"image": Binaries::toBase64(payload.attachments[0]),
+...
+}
+----
+
+== Troubleshoot Design Time WSDL Issues
+
+Some errors related to incorrect WSDL recognition, acquisition, or parsing may lead to some of the following issues:
+
+* DataSense does not correctly expose the message metadata defined in the WSDL.
+* In the Web Service Consumer Connector configuration screen, the *Operations* field does not show any Web service operation names that the connector aims to invoke:
++
+.Web Service Consumer Connector Operations field
+image::wsc-troubleshoot-1.png[Web Service Consumer Connector Operations field shows empty]
+
+* In the Web Service Consumer Connector global element configuration, the *Port* and *Address* fields are not automatically filled when a WSDL location is provided.
++
+.Web Service Consumer Connector Global Element
+image::wsc-troubleshoot-2.png[In the Web Service Consumer Connector global element window, the port and address fields are not filled when a WSDL location is selected]
+
+To resolve these issues:
+
+* Check the integrity of your WSDL by using your preferred online or desktop viewer applications. +
+Every parsing error during the configuration it not shown at design time, so an external application is needed.
```

Comment:
```suggestion
Because every parsing error during the configuration is not shown during design, implementing an external application helps detect them.
```